### PR TITLE
ref(ts): Convert groupActivity to Typescript

### PR DESCRIPTION
**Pull request by taylangocmen**

### CODE REVIEW

File: src/sentry/static/sentry/app/views/organizationGroupDetails/groupActivity.tsx [hunk 1]
```diff
@@ -15,27 +15,35 @@ import ConfigStore from 'app/stores/configStore';
...
-function makeDefaultErrorJson() {
-  return {detail: t('Unknown error. Please try again.')};
-}
-
-class GroupActivity extends React.Component {
+type Props = {
+  api: Client;
+  organization: Organization;
+  group: Group;
+  params: Params;
```
Comment (scttcper) [inline src/sentry/static/sentry/app/views/organizationGroupDetails/groupActivity.tsx:31]: should this be the `& RouteComponentProps<{}, {}>;` we use other places?

File: src/sentry/static/sentry/app/components/activity/note/types.tsx [hunk 2]
```diff
@@ -20,10 +22,12 @@ export type Mentioned = [string, string];
...
 export type MentionChangeEvent = Parameters<OnChangeHandlerFunc>[0];
 
-export type CreateError = {
-  detail: {
-    message: string;
-    code: number;
-    extra: any;
-  };
-};
+export type CreateError =
```
Comment (priscilawebdev) [inline src/sentry/static/sentry/app/components/activity/note/types.tsx:25]: everywhere we use this type, we also use `null`, like` CreateError | null`. So I was wondering if we should also add null in this type 🧐  but I am only thinking out loud... this is something for another PR 😉

**Review by priscilawebdev — APPROVED** [review]
it looks good to me with @scttcper 's suggestion 😉

File: src/sentry/static/sentry/app/components/activity/note/types.tsx (hunk 2 above)
Comment (taylangocmen) [reply src/sentry/static/sentry/app/components/activity/note/types.tsx:25]: I thought about that too, that's more practical for sure but having it like this in the components' props is ver explicit and helps understand how this prop behaves.

**Review by markstory — APPROVED** [review]

**Pull request merged**